src/components/modals/TemplatesModal.tsx

```diff
@@ -0,0 +1,197 @@
+import { ExerciseActivity } from "@/models/exercise-activity.model";
+import { Modal } from "@mui/material";
+import React, { useState } from "react";
+import { FiEye, FiX } from "react-icons/fi";
+import { BasicRoundedButton } from "../buttons/basic-rounded-button/Basic-rounded-button";
+import TemplateDataModal from "./TemplateDataModal";
+
+interface TemplateModalProps {
+  open: boolean;
+  onClose: () => void;
+  onGenerate: () => void;
+  onTemplateSelect: (template: ExerciseActivity[] | null) => void;
+}
+
+let mockData = {
```

Comment:
small note: maybe mark with with a `//TODO:` comment just so we don't miss deleting this?